[tests] Change invalidblockrequest to use BitcoinTestFramework

jnewbery opened the pull request:
Builds on #11771. Please review that PR first

Next step in #10603.

- first commit tidies up invalidblockrequest.py
- second commit removes usage of ComparisonTestFramework

jnewbery committed:
[tests] Change invalidblockrequest to use BitcoinTestFramework

[tests] update tests from changes to mininode in #11771 - added by @conscott

[tests] trivial update to hex conversion for readability - added by @conscott

jnewbery committed:
[tests] Fix flake8 warnings in invalidblockrequest

jnewbery commented:
This requires rebase, but I'll leave it as it is until #11771 is reviewed/merged.

ryanofsky commented on test/functional/invalidblockrequest.py:
```diff
@@ -11,68 +11,64 @@
-            height += 1
-        yield test
+        best_block = node.getblock(node.getbestblockhash())
+        tip = int("0x" + node.getbestblockhash(), 0)
```
In commit "Change invalidblockrequest to use BitcoinTestFramework"

Maybe avoid 0x prefix here and above by calling int(hash, 16)

ryanofsky reviewed (commented):
Reviewed last two commits only. Other commits are part of #11771 (and out of date).

- utACK 62bb3a0892558cf8813a22e1bf7860598dde3441 [tests] Change invalidblockrequest to use BitcoinTestFramework
- utACK 1d26a167c73b3e94509b3d39fa5875fdb9601eb1 [tests] Fix flake8 warnings in invalidblockrequest

jnewbery commented:
This requires rebase now that #11771 has been merged.

Review comments from @conscott to address in this PR:

- [ ] unused update to `block_time` variable can be removed (https://github.com/bitcoin/bitcoin/pull/11771#discussion_r167710791)
- [ ] unused update to `height` variable can be removed (https://github.com/bitcoin/bitcoin/pull/11771#pullrequestreview-95970049)
- [ ] `send_txs_and_test()` is never called with `sucess=True`. Either remove that variable or add a successful call to `send_txs_and_test()` to exercise code path (https://github.com/bitcoin/bitcoin/pull/11771#pullrequestreview-95979452)

conscott committed:
[test] Fix nits leftover from 11771

Remove unused variable reassignments in p2p_invalid_tx.py and call
send_txs_and_test() with valid transaction.

jnewbery commented:
Thanks to @conscott for rebasing this and fixing up the remaining issues from #11771 and Russ's review comment.

jamesob commented:
:+1: 

utACK https://github.com/bitcoin/bitcoin/pull/11772/commits/e97b113b049fa1ca7152373d6ea911d56bf7ef4e

MarcoFalke commented:
utACK e97b113b049fa1ca7152373d6ea911d56bf7ef4e

jnewbery commented:
> Not an issue with your code, but __eq__() is not implemented for CTransaction, so the argument to assert is always true, since none of the tx objects are the same reference (they were copied).

Agree - this assert: https://github.com/bitcoin/bitcoin/blob/00d1680498c5550e7db1f359202d3433a092fafd/test/functional/p2p_invalid_block.py#L83 should be removed in a future PR.
